From 8a03e56b253e9691c90bc52ca199323d71b96204 Mon Sep 17 00:00:00 2001
Message-Id: <8a03e56b253e9691c90bc52ca199323d71b96204.1648732216.git.geliang.tang@suse.com>
From: Pawan Gupta <pawan.kumar.gupta@linux.intel.com>
Date: Fri, 29 Oct 2021 12:43:54 -0700
Subject: [PATCH] bpf: Disallow unprivileged bpf by default
Git-commit: 8a03e56b253e9691c90bc52ca199323d71b96204
Patch-mainline: v5.16-rc1
References: bsn#127
Modified-by-SEL: Yes, refreshed due to context changes


Disabling unprivileged BPF would help prevent unprivileged users from
creating certain conditions required for potential speculative execution
side-channel attacks on unmitigated affected hardware.

A deep dive on such attacks and current mitigations is available here [0].

Sync with what many distros are currently applying already, and disable
unprivileged BPF by default. An admin can enable this at runtime, if
necessary, as described in 08389d888287 ("bpf: Add kconfig knob for
disabling unpriv bpf by default").

  [0] "BPF and Spectre: Mitigating transient execution attacks", Daniel Borkmann, eBPF Summit '21
      https://ebpf.io/summit-2021-slides/eBPF_Summit_2021-Keynote-Daniel_Borkmann-BPF_and_Spectre.pdf

Signed-off-by: Pawan Gupta <pawan.kumar.gupta@linux.intel.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Mark Rutland <mark.rutland@arm.com>
Link: https://lore.kernel.org/bpf/0ace9ce3f97656d5f62d11093ad7ee81190c3c25.1635535215.git.pawan.kumar.gupta@linux.intel.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/Kconfig |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/kernel/bpf/Kconfig
+++ b/kernel/bpf/Kconfig
@@ -79,6 +79,12 @@ config BPF_UNPRIV_DEFAULT_OFF
 
 	  If you are unsure how to answer this question, answer Y.
 
+	  Unprivileged BPF could be used to exploit certain potential
+	  speculative execution side-channel vulnerabilities on unmitigated
+	  affected hardware.
+
+	  If you are unsure how to answer this question, answer Y.
+
 source "kernel/bpf/preload/Kconfig"
 
 config BPF_LSM
